akka-typed/src/main/scala/akka/typed/ActorContext.scala

```diff
@@ -0,0 +1,247 @@
+
+  /**
+   * Create a child Actor from the given [[Props]] and with the given name.
+   */
+  def spawn[U](props: Props[U], name: String): ActorRef[U]
+
+  /**
+   * Create an untyped child Actor from the given [[akka.actor.Props]] under a randomly chosen name.
+   * It is good practice to name Actors wherever practical.
+   */
+  def actorOf(props: a.Props): a.ActorRef
+
+  /**
+   * Create an untyped child Actor from the given [[akka.actor.Props]] and with the given name.
+   */
+  def actorOf(props: a.Props, name: String): a.ActorRef
+
+  /**
+   * Force the child Actor under the given name to terminate after it finishes
+   * processing its current message. Nothing happens if no Actor exists with the
+   * given name.
+   */
+  def stop(childName: String): Unit
+
+  /**
+   * Register for [[Terminated]] notification once the Actor identified by the
+   * given [[ActorRef]] terminates. This notification is also generated when the
+   * [[ActorSystem]] to which the referenced Actor belongs is declared as
+   * failed (e.g. in reaction to being unreachable).
+   */
+  def watch[U](other: ActorRef[U]): ActorRef[U]
+
+  /**
+   * Register for [[Terminated]] notification once the Actor identified by the
+   * given [[akka.actor.ActorRef]] terminates. This notification is also generated when the
+   * [[ActorSystem]] to which the referenced Actor belongs is declared as
+   * failed (e.g. in reaction to being unreachable).
+   */
+  def watch(other: akka.actor.ActorRef): akka.actor.ActorRef
+
+  /**
+   * Revoke the registration established by [[#watch[U]* watch]]. A [[Terminated]]
+   * notification will not subsequently be received for the referenced Actor.
+   */
+  def unwatch[U](other: ActorRef[U]): ActorRef[U]
+
+  /**
+   * Revoke the registration established by [[#watch(* watch]]. A [[Terminated]]
+   * notification will not subsequently be received for the referenced Actor.
+   */
+  def unwatch(other: akka.actor.ActorRef): akka.actor.ActorRef
+
+  /**
+   * Schedule the sending of a [[ReceiveTimeout]] notification in case no other
+   * message is received during the given period of time. The timeout starts anew
+   * with each received message. Provide `Duration.Undefined` to switch off this
+   * mechanism.
+   */
+  def setReceiveTimeout(d: Duration): Unit
+
+  /**
+   * When handling a [[Failed]] signal the behavior should invoke this method to
+   * inform the execution mechanism of the supervisor decision. If this is not
+   * done then the implied decision is to escalate the failure.
+   */
+  def setFailureResponse(decision: Failed.Decision): Unit
+
+  /**
+   * Schedule the sending of the given message to the given target Actor after
+   * the given time period has elapsed. The scheduled action can be cancelled
+   * by invoking [[akka.actor.Cancellable!.cancel* cancel]] on the returned
+   * handle.
+   */
+  def schedule[U](delay: FiniteDuration, target: ActorRef[U], msg: U): a.Cancellable
```

Comment:
Actors react to two things: messages and timeouts. Since they can only receive messages, timeouts need to be messages as well.

There is an inherent unfixable race between cancellation and expiry, leading to messages being “canceled” while they are in the mailbox already; if someone has a smart idea how we can safely and with low overhead make those invisible to the behavior I’m all ears, but even without this fix I do think that delayed messages are a fundamental operation in Actor systems.